subsquid/src/pabloProcessor.ts

```diff
@@ -0,0 +1,299 @@
+            .find((asset) => asset.assetId == pool.quoteAssetId);
+        if (quoteAsset == undefined) {
+            throw new Error('quoteAsset not found');
+        }
+        if (isReverse) {
+            console.debug('Reverse swap');
+            // volume
+            pool.totalVolume = Big(pool.totalVolume).add(Big(swappedEvt.baseAmount.toString())).toString();
+            baseAsset.totalVolume += swappedEvt.quoteAmount;
+            quoteAsset.totalVolume += swappedEvt.baseAmount;
+
+            // liquidity
+            pool.totalLiquidity = Big(pool.totalLiquidity)
+                // fees TODO is this correct?
+                .sub(Big(swappedEvt.fee.toString()))
+                .toString();
+            // for reverse exchange "default quote" (included as the base amount in the evt) amount leaves the pool
+            baseAsset.totalLiquidity += swappedEvt.quoteAmount;
+            quoteAsset.totalLiquidity -= swappedEvt.baseAmount;
+            quoteAsset.totalLiquidity -= swappedEvt.fee;
+         } else {
+            console.debug('Normal swap');
+            // volume
+            pool.totalVolume = Big(pool.totalVolume).add(Big(swappedEvt.quoteAmount.toString())).toString();
+            baseAsset.totalVolume += swappedEvt.baseAmount;
+            quoteAsset.totalVolume += swappedEvt.quoteAmount;
+
+            // liquidity
+            pool.totalLiquidity = Big(pool.totalLiquidity)
+                // fees TODO is this correct?
+                .sub(
+                    // calculated the quote amount based on the exchange rate as the fees are in the base asset
+                    Big(swappedEvt.quoteAmount.toString())
+                        .div(Big(swappedEvt.baseAmount.toString()))
+                        .mul(Big(swappedEvt.fee.toString())))
+                .toString();
+            // for normal exchange "default quote" amount gets into the pool
+            baseAsset.totalLiquidity -= swappedEvt.baseAmount;
+            baseAsset.totalLiquidity -= swappedEvt.fee;
+            quoteAsset.totalLiquidity += swappedEvt.quoteAmount;
+        }
+        baseAsset.calculatedTimestamp = timestamp;
+        baseAsset.blockNumber = BigInt(ctx.block.height);
+        quoteAsset.calculatedTimestamp = timestamp;
+        quoteAsset.blockNumber = BigInt(ctx.block.height);
+
+        let tx = await get(ctx.store, PabloTransaction, ctx.event.id);
+        if (tx != undefined) {
+            throw new Error("Unexpected transaction in db");
+        }
+        tx = createTransaction(ctx, pool, who,
+            PabloTransactionType.SWAP,
+            isReverse
+                ? Big(swappedEvt.baseAmount.toString()).div(Big(swappedEvt.quoteAmount.toString())).toString()
+                : Big(swappedEvt.quoteAmount.toString()).div(Big(swappedEvt.baseAmount.toString())).toString(),
+            BigInt(baseAsset.assetId),
+            swappedEvt.baseAmount,
+            pool.quoteAssetId,
+            swappedEvt.quoteAmount);
```

Comment:
shouldn't we need to check for `isReverse` here?

Reply:
ah no this is meant to capture the base/quote as it was emitted by the event to not lose that information. As you may see, each transaction contains the base/quote assetIds.